src/wasm/wasm-validator.cpp

```diff
@@ -797,6 +797,10 @@ void FunctionValidator::visitCallIndirect(CallIndirect* curr) {
   if (curr->target->type != Type::unreachable) {
     auto* table = getModule()->getTableOrNull(curr->table);
     shouldBeTrue(!!table, curr, "call-indirect table must exist");
+    shouldBeSubType(table->type,
+                    Type::funcref,
+                    curr,
+                    "call-indirect table type must be a funcref.");
```

Comment:
Could we allow `table->type.isFunction()` here? More specific function types are ok unless I'm missing something.

Reply:
You're right, I had totally missed this. I'll add a test case for it as well. Thanks!

Reply:
Oh, I missed that it checked for a subtype... yeah, it's correct, but would be clearer the other way, I agree

Reply:
It's now using `->isFunction()`.